src/middleware/aws.rs

```diff
@@ -0,0 +1,98 @@
+use opentelemetry::{
+    global,
+    trace::{SpanKind, Tracer},
+};
+pub use opentelemetry::{trace::Span, KeyValue};
+pub use opentelemetry_semantic_conventions as semcov;
+
+// TODO: Write as macro
+//
+// #[instrument_aws(table_name = "cars", operation = "CreateCar", method = "Post")]
+// fn create_car_in_database() {
+//      info_span_dynamo();
+// }
+
+// Once this scope is closed, all spans inside are closed as well
+#[cfg(any(feature = "aws", feature = "aws_dynamo"))]
+pub fn info_span_dynamo(
+    table_name: &str,
+    operation: &str,
+    method: &str,
+    parent_context: &opentelemetry::Context,
+) -> opentelemetry::global::BoxedSpan {
+    // Spans will be sent to the configured OpenTelemetry exporter
+
+    let tracer = global::tracer("aws_sdk");
+    let mut span = tracer
+        .span_builder("aws_dynamo")
+        .with_attributes(vec![
+            semcov::trace::RPC_METHOD.string(method.to_string()),
+            semcov::trace::RPC_SYSTEM.string("aws-api"),
+            semcov::trace::RPC_SERVICE.string("DynamoDB"),
```

Comment:
You could import those constants instead of prefixing them with `semcov::trace:: ` every time:

```rust
pub use opentelemetry_semantic_conventions::trace::{
    RPC_METHOD, RPC_SERVICE, RPC_SYSTEM,
};
```

But there is no real difference, so use whatever method you prefer.